parachain/pallets/control/src/lib.rs

```diff
@@ -282,13 +285,14 @@ pub mod pallet {
 		) -> DispatchResult {
 			let location: MultiLocation = T::AgentOrigin::ensure_origin(origin)?;
 
-			let (agent_id, _, _) = Self::convert_location(location)?;
+			let (agent_id, para_id, location) = Self::convert_location(location)?;
 
 			ensure!(Agents::<T>::contains_key(agent_id), Error::<T>::AgentNotExist);
 
 			let message = Message {
-				origin: T::OwnParaId::get(),
+				origin: para_id,
```

Comment:
For `UpdateChannel` and `TransferNativeFromAgent` chage the origin to the original Parachain, not bridgeHub.